velox/type/TimestampConversion.cpp

```diff
@@ -593,6 +613,28 @@ Status daysSinceEpochFromWeekDate(
   return Status::OK();
 }
 
+Expected<int64_t> daysSinceEpochFromWeekOfMonthDate(
+    int32_t year,
+    int32_t month,
+    int32_t weekOfMonth,
+    int32_t dayOfWeek) {
+  if (!isValidWeekOfMonthDate(year, month, weekOfMonth, dayOfWeek)) {
+    return folly::makeUnexpected(Status::UserError(
+        "Date out of range: {}-{}-{}-{}", year, month, weekOfMonth, dayOfWeek));
+  }
+  int64_t daysSinceEpochOfFirstDayOfMonth;
+  Status status =
+      daysSinceEpochFromDate(year, month, 1, daysSinceEpochOfFirstDayOfMonth);
+  if (status.ok() == false) {
+    return folly::makeUnexpected(status);
+  }
+  int32_t firstDayOfWeek =
+      extractISODayOfTheWeek(daysSinceEpochOfFirstDayOfMonth);
```

Comment:
I notice `extractISODayOfTheWeek` casts input to int64_t. How about changing parameter to int64_t type instead of int32_t type

Reply:
Looks like the cast is to avoid overflow in addition and negating. If we change it as int64_t, overflow needs to considered as well by casting to a larger type.

Reply:
Thanks for your explanation. I see `daysSinceEpochFromWeekDate` function also calls this funtion with a int64_t argument. I think we change `extractISODayOfTheWeek` parameter to int64_t and cast it to int128_t when do calculate would be better. How do you think?

Reply:
I wonder if there is any limitation for year? If the days cannot exceed INT32_MAX, it would be safe to call it.

Reply:
The max year is [292278994](https://github.com/facebookincubator/velox/blob/main/velox/type/TimestampConversion.h#L42) so the days can be (292278994-1970)*365=106,681,113,760 that is bigger than 2,147,483,647(INT32_MAX).

Reply:
> I think we change extractISODayOfTheWeek parameter to int64_t and cast it to int128_t when do calculate

Looks fine to me. @mbasmanova What are your thoughts? Thanks.